contracts/meta-oracles/LinearizedPriceDataSource.sol

```diff
@@ -0,0 +1,199 @@
+import { SafeMath } from "openzeppelin-solidity/contracts/math/SafeMath.sol";
+
+import { IMedian } from "../external/DappHub/interfaces/IMedian.sol";
+import { IDataFeed } from "./interfaces/IDataFeed.sol";
+
+
+/**
+ * @title LinearizedPriceDataSource
+ * @author Set Protocol
+ *
+ * This DataSource returns the current value of the Medianizer Oracle. If the interpolationThreshold
+ * is reached, then returns a linearly interpolated value.
+ * It is intended to be read by a DataFeed smart contract.
+ */
+contract LinearizedPriceDataSource is
+    Ownable
+{
+    using SafeMath for uint256;
+
+    /* ============ State Variables ============ */
+    uint256 public interpolationThreshold; 
+    string public dataDescription;
+    IMedian public medianizerInstance;
+
+    /* ============ Events ============ */
+
+    event LogMedianizerUpdated(
+        address newMedianizerAddress
+    );
+
+    /* ============ Constructor ============ */
+
+    /*
+     * Set interpolationThreshold, data description, and instantiate medianizer
+     *
+     * @param  _interpolationThreshold    The minimum time in seconds where interpolation is enabled
+     * @param  _medianizerAddress         The address to read current data from
+     * @param  _dataDescription           Description of contract for Etherscan / other applications
+     */
+    constructor(
+        uint256 _interpolationThreshold,
+        address _medianizerAddress,
+        string memory _dataDescription
+    )
+        public
+    {
+        interpolationThreshold = _interpolationThreshold;
+        medianizerInstance = IMedian(_medianizerAddress);
+        dataDescription = _dataDescription;
+    }
+
+    /* ============ External ============ */
+
+    /*
+     * Returns the data from the Medianizer contract. If the current timestamp has surpassed
+     * the interpolationThreshold, then the current price is retrieved and interpolated based on
+     * the previous value and the time that has elapsed since the intended update value.
+     * Note: The sender must be a DataSource contract.
+     *
+     * Returns with newest data point by querying medianizer. Is eligible to be
+     * called after nextAvailableUpdate timestamp has passed. Because the nextAvailableUpdate occurs
+     * on a predetermined cadence based on the time of deployment, delays in calling poke do not propogate
+     * throughout the whole dataset and the drift caused by previous poke transactions not being mined
+     * exactly on nextAvailableUpdate do not compound as they would if it was required that poke is called
+     * an updatePeriod amount of time after the last poke.
+     *
+     * @returns                Returns the datapoint from the Medianizer contract
+     */
+    function read()
+        external
+        returns (uint256)
+    {
+        uint256 nextEarliestUpdate = IDataFeed(msg.sender).nextEarliestUpdate();
+
+        // Make sure block timestamp exceeds nextEarliestUpdate
```

Comment:
Why have this check on the DataSource and not the DataFeed. Seems like time based checks should be on the DataFeed since that's where we're concerned with maintaining the integrity of the data's cadence. The DataSource is more for querying the correct price.

Reply:
+1